faur.1
.\" Automatically generated by Pandoc 2.2.2
.\"
.TH "FAUR" "1" "July 17, 2018" "\- faur\-0.1" ""
.hy
.SH NAME
.PP
faur \- lightweight, opinionated AUR helper
.SH SYNOPSIS
.PP
faur fetch [\f[I]pkgs\f[]]\&...
.PP
faur update
.PP
faur help
.SH DESCRIPTION
.PP
faur downloads packages to the directory specified in $AURDIR.
If $AURDIR is unset $HOME/aur is used as a default.
Packages are cloned from the git repository located under $AUR_PATH, if
unset https://aur.archlinux.org/ is used as a default value.
.PP
The update command checks to see which if any of the packages underneath
$AURDIR need to be updated and notifies the user.
.SH DEPENDENCY MANAGEMENT
.PP
faur \f[I]does not\f[] attempt to resolve dependencies, as doing so
would discourage the user from reading PKGBUILD files before
installation.
.PP
Users should bear in mind that the AUR is an \f[I]untrusted\f[] source
of packages and always exercise caution when installing packages from
it.
.SH AUTHORS
Joe Davis <me@jo.ie>.


NEWS
2016-06-17  Kamil Prusko  <kamilprusko@gmail.com>
	Release version 0.12.1:

	* Fix for build error

2016-06-11  Kamil Prusko  <kamilprusko@gmail.com>
	Release version 0.12.0:

	* Support for more desktops
	* Ability to pause the timer
	* Ability add plugins
	* Code refactoring
	* Bug fixes
	* Dropped Telepathy and Skype integration
	* Dropped support for GNOME Shell 3.14 and older

2016-05-06  Kamil Prusko  <kamilprusko@gmail.com>
	Release version 0.11.3:

	* Ignore small mouse movements in screen notification
	* Fixed indicator not updating
	* Cleaned up some compilation warnings (thanks @aerostitch)

2016-03-24  Kamil Prusko  <kamilprusko@gmail.com>
	Release version 0.11.2:

	* Support for GNOME Shell 3.20

2015-10-24  Kamil Prusko  <kamilprusko@gmail.com>
	Release version 0.11.1:

	* Support for GNOME Shell 3.18
	* Turkish translation (thanks @ErkanMDR)

2015-06-13  Kamil Prusko  <kamilprusko@gmail.com>
	Release version 0.11.0:

	* Support for GNOME Shell 3.14 and 3.16
	* Ability to change indicator appearance
	* Skype and Epiphany integration
	* Updated French translation (thanks @neitsab)
	* Code refactoring
	* Basic unit tests
	* Bug fixes

2014-10-17  Kamil Prusko  <kamilprusko@gmail.com>
	Release version 0.10.3:

	* Support for GNOME Shell 3.10, 3.12 and 3.14
	* Updated Catalan translation (thanks @Ecron)
	* Code cleanups (thanks @aerostitch)
	* Bug fixes

2014-06-13  Kamil Prusko  <kamilprusko@gmail.com>

	Release version 0.10.2:

	* No need to restart gnome-shell to enable extension

2014-05-02  Kamil Prusko  <kamilprusko@gmail.com>

	Release version 0.10.1:

	* Support for GNOME Shell 3.10 and 3.12
	* Louder sounds
	* Fixed brining preferences dialog to focus
	* Fixed change of notications volume
	* Fixed reminders being showed up during pomodoro

2014-01-11  Kamil Prusko  <kamilprusko@gmail.com>

	Release version 0.10.0:

	* Support for GNOME Shell 3.10
	* New layout in preferences dialog
	* Migrate to gsteramer-1.0
	* Updated translations

2013-12-30  Kamil Prusko  <kamilprusko@gmail.com>

	Release version 0.9.1:

	* Support for GNOME Shell 3.8
	* Improved long pause scheduling
	* Deactivate screensaver to notify start of pomodoro
	* German translation (thanks @linuxrider)
	* Bug fixes

2013-11-17  Kamil Prusko  <kamilprusko@gmail.com>

	Release version 0.9.0:

	* Support for GNOME Shell 3.8
	* Added a preferences dialog
	* Improved timer accuracy
	* Bug fixes

2013-10-18  Kamil Prusko  <kamilprusko@gmail.com>

	Release version 0.8.1:

	* Support for GNOME Shell 3.10

2013-04-28  Kamil Prusko  <kamilprusko@gmail.com>

	Release version 0.8:

	* Support for GNOME Shell 3.8 (thanks @haaja)
	* Brazilian Portuguese translation (thanks @aleborba)
	* Minor bug fixes

2013-01-15  Kamil Prusko  <kamilprusko@gmail.com>

	Release version 0.7:

	* Support for GNOME Shell 3.4 and 3.6
	* Feature: Full screen notifications
	* Feature: Reminders
	* Chinese translation (thanks @mengzhuo)
	* Czech translation (thanks @veverjak)

2012-04-13  Arun Mahapatra  <pratikarun@gmail.com>

	Release version 0.6:

	* Support for GNOME Shell 3.4
	* Breaking change: Dropped support for older gnome-shell versions due to
	incompatible APIs
	* Feature: Support for "Away from desk" mode
	* Feature: Ability to change IM presence status based on pomodoro activity
	* New translation: Persian (thanks @arashm)
	* Fixed issues #38, #39, #41, #42, #45 and
	[more](https://github.com/codito/gnome-shell-pomodoro/issues?sort=created&direction=desc&state=closed&page=1)

2011-12-04  Arun  <pratikarun@gmail.com>

	Release version 0.5:

	* Bunch of cleanups, user interface awesomeness [Issue #37, Patch from @kamilprusko]
	* Config options are changed to more meaningful names [above patch]

2011-11-20  Arun  <pratikarun@gmail.com>

	Release version 0.4:

	* Sound notification at end of a pomodoro break [Issue #26, Patch from @kamilprusko]
	* System wide config file support [Patch from @mgrela]
	* Support to skip breaks in case of persistent message [Patch from @amanbh]
	* Some minor bug fixes, and keybinder3 requirement is now optional


faur.sh
#!/bin/sh
# Copyright (c) 2018, Joe Davis <me@jo.ie>
#
# Permission to use, copy, modify, and/or distribute this software for any
# purpose with or without fee is hereby granted, provided that the above
# copyright notice and this permission notice appear in all copies.
#
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

set -e

AURDIR=${AURDIR:-$HOME/aur}
AUR_PATH=${AUR_PATH:-https://aur.archlinux.org/}

PROG=$(basename "$0")

usage() {
	printf 'USAGE: %s fetch [pkg...]\n' "$PROG" 2>&1
	printf '       %s update\n' "$PROG" 2>&1
	printf '       %s help\n' "$PROG" 2>&1
	exit $1
}

update() {
	printf "The following packages need updating: "
	none="none found"

	for pkg in $*; do
		out=$(cd "$pkg" && git pull)
		if [ "$out" != "Already up to date." ]; then
			none=''
			printf '\n  %s' "$pkg"
		fi
	done
	echo $none
}

fetch() {
	while [ $# -gt 0 ]; do
		git clone "${AUR_PATH}$1.git"
		shift
	done
}

if [ $# -eq 0 ]; then
	usage 1
fi

if [ "$1" = "help" ]; then
	usage 0
fi

cd "$AURDIR"

if [ "$1" = "fetch" ]; then
	shift
	if [ $# -eq 0 ]; then
		usage 1
	fi
	fetch $*
elif [ "$1" = "update" ]; then
	# Skip folders containing spaces, newlines and those beginning with '.'
	# This avoids problems with for loops over find output
	PKGS=$(find . -maxdepth 1 \
		! -name ".*" \
		! -name "* *" ! -name "$(printf "*\\n*")" \
		-type d)
	update $PKGS
else
	echo "Unknown command: $1" 2>&1
	usage
	exit 1
fi
